docs/autoupdate_overview.txt

```diff
@@ -0,0 +1,90 @@
+Dolphin's autoupdate procedure is spread through a number of files; this overview describes the
```

Comment:
It might be worth converting this to Markdown? "General notes", "Class and file responsibilities", "AutoUpdateChecker" etc. would naturally be section headers.